internal/hns/hns.go

```diff
@@ -15,10 +78,71 @@ func (e EndpointNotFoundError) Error() string {
 	return fmt.Sprintf("Endpoint %s not found", e.EndpointName)
 }
 
+// NetworkNotFoundError happens when users attempt to Read/Modify a Network that does not exist.
 type NetworkNotFoundError struct {
 	NetworkName string
 }
 
 func (e NetworkNotFoundError) Error() string {
 	return fmt.Sprintf("Network %s not found", e.NetworkName)
 }
+
+// HcnError happens when a call to a V2 Api returns an error.
+type HcnError struct {
+	methodName string
+	hr error
+	result string
+}
+
+func (e HcnError) Error() string {
+	errorOutput := fmt.Sprintf("%s - ", e.methodName)
+	if e.hr != nil {
+		errorOutput += fmt.Sprintf("HRESULT: %d ", e.hr)
+	}
+
+	if e.result != "" {
+		errorOutput += fmt.Sprintf("Result: %s", e.result)
+	}
+	
+	return errorOutput
+}
+
+func CheckForErrors(methodName string, hr error, resultBuffer *uint16) error {
+	errorFound := false
+	
+	if hr != nil {
+		errorFound = true
+	}
+
+	result := ""
+	if resultBuffer != nil {
+		result = interop.ConvertAndFreeCoTaskMemString(resultBuffer)
+		if result != "" {
+			errorFound = true
+		}
+	}
+
+	if errorFound {
+		err := HcnError{methodName: methodName, hr: hr, result: result}
+		return err
+	}
+
+	return nil
+}
+
+// PlatformDoesNotSupportError happens when users are attempting to use a newer shim on an older OS
+type PlatformDoesNotSupportError struct {
+	FeatureName string
+}
+
+func (e PlatformDoesNotSupportError) Error() string {
+	return fmt.Sprintf("Platform does not support feature %s", e.FeatureName)
```

Comment:
Minor nit - golang convention is that errors should start lower-case, so `platform does not support....`. There's a few of these.